### PR TITLE
fix: Exclude transitive dependencies of espresso-contrib

### PR DESCRIPTION
**Pull request by grzegorz-jarosz**
This PR takes forward work started by @tinder-ktarasov in his PR #497, hoping that this will fix issues in #449.

Initial submission had some failing tests (but logs are no longer accessible). 

In a short setting "transitive = false" stops gradle from loading espresso-contrib's dependendencies. However, adding it fails one of the tests with some "Navigation" class missing. After adding "com.google.android.material:material:1.2.0" this test no longer failed.

Note that android.material package is in espresson-contrib dependencies, so in fact we're removing all but one of dependent packages from build.

I'm not sure if that's the right "gradle" solution. I'm open for a discussion.

### CODE REVIEW

File: espresso-server/app/build.gradle [hunk 1]
```diff
@@ -89,8 +89,10 @@ dependencies {
     testImplementation "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
 
     androidTestImplementation "androidx.annotation:annotation:$annotation_version"
-    androidTestImplementation "androidx.test.espresso:espresso-contrib:$espresso_version"
-    androidTestImplementation "androidx.test.espresso:espresso-core:$espresso_version"
+    androidTestImplementation("androidx.test.espresso:espresso-contrib:$espresso_version") {
+        transitive = false
```
Comment (mykola-mokhnach) [inline espresso-server/app/build.gradle:93]: I would say it makes sense to add a comment here or a link the actual PR with a short explanation about why transitive deps are disabled

**Comment by grzegorz-jarosz** [comment]
Somehow, I knew it won't be that easy. Okay, back to the drawing board

**Comment by mykola-mokhnach** [comment]
> Somehow, I knew it won't be that easy. Okay, back to the drawing board

What exactly is complicated? The PR itself is ok

**Comment by grzegorz-jarosz** [comment]
Indeed, PR looks simple, but tests are failing for SDK_30 (when I run tests locally they work) and SDK_23 (haven't looked at that one yet)

**Comment by mykola-mokhnach** [comment]
they were failing even before. It looks like it's partially an azure issue (the emulator is just too slow and times out unexpectedly)

**Comment by mykola-mokhnach** [comment]
simply add the comment and I would be happy to merge the PR

**Comment by grzegorz-jarosz** [comment]
I've added a comment with short description. 

This may not finish all the "Drawable and Lifecycle" issues as there is other PR (#506 ) that took this work further.

In one case I had to use `"additionalAppDependencies": ["com.google.android.material:material:1.2.0"]` in espressoBuildConfig to solve some issues I had in my case. If I remember correctly those dependencies go into the gradle as "implements".

I guess I'm trying to say that there is a room for improvements.

File: espresso-server/app/build.gradle (hunk 1 above)
Comment (grzegorz-jarosz) [reply espresso-server/app/build.gradle:95]: Added in f9b162213e159ac2c510a42d8c47a1b097c4f1c9.

**Comment by mykola-mokhnach** [comment]
> I guess I'm trying to say that there is a room for improvements.

There always is. Lets do it in small steps.

**Pull request merged**